### PR TITLE
Add supports for multiple TokenLookup

### LINKED ISSUE #55: TokenLookup accepts multiple methods
Is it a good practice to allow the lookup having multiple methods instead of a single method only ?

### PR DESCRIPTION
**Pull request by changkun**
resolve #55

### CODE REVIEW

**Comment by changkun** [comment]
@appleboy Are you still maintaining this project? Otherwise, I want to close this PR asap.

**Comment by appleboy** [comment]
yes.

**Comment by changkun** [comment]
@appleboy Then will you approve this PR or reason for rejection?

This feature enables clients do not limit to send jwt in one location, but middleware handles for different locations, such as if jwt was missing in the header then lookup for request body or cookies.

**Comment by appleboy** [comment]
Please see the errors on drone and help to fix.

http://drone.wu-boy.com/appleboy/gin-jwt/100/4

File: auth_jwt.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 	"time"
 
 	"github.com/gin-gonic/gin"
-	"gopkg.in/dgrijalva/jwt-go.v3"
+	jwt "gopkg.in/dgrijalva/jwt-go.v3"
```
Comment (appleboy) [inline auth_jwt.go]: please ignore this changes.

**Comment by changkun** [comment]
@appleboy The test should be passed, I don't understand why the CI testing doesn't pass but my local testing gracefully passed everything.

**Comment by appleboy** [comment]
I will take it.

File: auth_jwt.go [hunk 2]
```diff
@@ -528,14 +528,25 @@ func (mw *GinJWTMiddleware) parseToken(c *gin.Context) (*jwt.Token, error) {
...
-	parts := strings.Split(mw.TokenLookup, ":")
-	switch parts[0] {
-	case "header":
-		token, err = mw.jwtFromHeader(c, parts[1])
-	case "query":
-		token, err = mw.jwtFromQuery(c, parts[1])
-	case "cookie":
-		token, err = mw.jwtFromCookie(c, parts[1])
+	methods := strings.Split(mw.TokenLookup, ",")
+	for _, method := range methods {
```
Comment (appleboy) [inline auth_jwt.go]: Add the folowing check.

```go
if len(token) > 0 {
	break
}
```

File: auth_jwt.go [hunk 3]
```diff
@@ -528,14 +528,25 @@ func (mw *GinJWTMiddleware) parseToken(c *gin.Context) (*jwt.Token, error) {
...
+		parts := strings.Split(strings.TrimSpace(method), ":")
+		k := strings.TrimSpace(parts[0])
+		v := strings.TrimSpace(parts[1])
+		switch k {
+		case "header":
+			if len(token) == 0 {
```
Comment (appleboy) [inline auth_jwt.go]: remove `if` block.

File: auth_jwt.go [hunk 4]
```diff
@@ -528,14 +528,25 @@ func (mw *GinJWTMiddleware) parseToken(c *gin.Context) (*jwt.Token, error) {
...
+				token, err = mw.jwtFromHeader(c, v)
+			}
+		case "query":
+			if len(token) == 0 {
```
Comment (appleboy) [inline auth_jwt.go]: same as above

File: auth_jwt.go [hunk 5]
```diff
@@ -528,14 +528,25 @@ func (mw *GinJWTMiddleware) parseToken(c *gin.Context) (*jwt.Token, error) {
...
+				token, err = mw.jwtFromQuery(c, v)
+			}
+		case "cookie":
+			if len(token) == 0 {
```
Comment (appleboy) [inline auth_jwt.go]: same as above

**Pull request merged**